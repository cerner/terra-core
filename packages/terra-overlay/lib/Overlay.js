'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Portal = require('./Portal');

var _Portal2 = _interopRequireDefault(_Portal);

var _OverlayManager = require('./OverlayManager');

var _OverlayManager2 = _interopRequireDefault(_OverlayManager);

require('./Overlay.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var ESCAPE = 27;
var overlayManager = new _OverlayManager2.default();

var propTypes = {
  /**
   * Whether or not <Overlay/> should be rendered
   */
  isOpen: _react.PropTypes.bool,
  /**
   * Set full screen overlay
   */
  isFullScreen: _react.PropTypes.bool,
  /**
   * DOM element that will be overlaid
   */
  container: _react.PropTypes.object,
  /**
   * Callback function for closing the overlay
   */
  onRequestClose: _react.PropTypes.func,
  /**
   * Allow closing overlay by clicking backdrop
   */
  canClickOutsideClose: _react.PropTypes.bool,
  /**
   * Center the content horizontally and vertically
   */
  centerContent: _react.PropTypes.bool,
  /**
   * Visible overlay content when overlay is open
   */
  children: _react.PropTypes.node
};

var defaultProps = {
  isOpen: false,
  isFullScreen: true,
  container: document.body,
  centerContent: true,
  canClickOutsideClose: true
};

var Overlay = function (_React$Component) {
  _inherits(Overlay, _React$Component);

  function Overlay() {
    _classCallCheck(this, Overlay);

    var _this = _possibleConstructorReturn(this, (Overlay.__proto__ || Object.getPrototypeOf(Overlay)).call(this));

    _this.state = {};
    _this.trapFocus = _this.trapFocus.bind(_this);
    _this.handleEscKeyup = _this.handleEscKeyup.bind(_this);
    _this.handleClickOutside = _this.handleClickOutside.bind(_this);
    return _this;
  }

  _createClass(Overlay, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      if (this.props.isOpen) {
        this.onOpen();
      }
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate(prevProps) {
      if (!prevProps.isOpen && this.props.isOpen) {
        this.onOpen();
      } else if (prevProps.isOpen && !this.props.isOpen) {
        this.onClose();
      }
    }
  }, {
    key: 'onOpen',
    value: function onOpen() {
      overlayManager.add(this, this.props.container);
      this.focus();
      document.addEventListener('focus', this.trapFocus, true);
      document.addEventListener('keyup', this.handleEscKeyup);
    }
  }, {
    key: 'onClose',
    value: function onClose() {
      overlayManager.remove(this, this.props.container);
      document.removeEventListener('focus', this.trapFocus, true);
      document.removeEventListener('keyup', this.handleEscKeyup);
      this.restoreFocus();
    }
  }, {
    key: 'focus',
    value: function focus() {
      this.focusBeforeOpen = document.activeElement;
      this.overlay.firstElementChild.focus();
    }
  }, {
    key: 'trapFocus',
    value: function trapFocus() {
      var currentActive = document.activeElement;
      var child = this.overlay.firstElementChild;
      if (child !== currentActive && !child.contains(currentActive)) {
        child.focus();
      }
    }
  }, {
    key: 'restoreFocus',
    value: function restoreFocus() {
      if (this.focusBeforeOpen && this.focusBeforeOpen.focus) {
        this.focusBeforeOpen.focus();
        this.focusBeforeOpen = null;
      }
    }
  }, {
    key: 'handleEscKeyup',
    value: function handleEscKeyup(e) {
      if (e.keyCode === ESCAPE) {
        this.props.onRequestClose();
      }
    }
  }, {
    key: 'handleClickOutside',
    value: function handleClickOutside(e) {
      if (this.props.canClickOutsideClose && e.target === e.currentTarget) {
        this.props.onRequestClose();
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props = this.props,
          isOpen = _props.isOpen,
          isFullScreen = _props.isFullScreen,
          container = _props.container,
          onRequestClose = _props.onRequestClose,
          canClickOutsideClose = _props.canClickOutsideClose,
          centerContent = _props.centerContent,
          children = _props.children,
          customProps = _objectWithoutProperties(_props, ['isOpen', 'isFullScreen', 'container', 'onRequestClose', 'canClickOutsideClose', 'centerContent', 'children']);

      var overlayClass = (0, _classnames2.default)('terra-Overlay', { 'terra-Overlay--center': centerContent }, customProps.className);
      var onELementPosition = isFullScreen ? null : {
        position: 'absolute',
        top: container.scrollTop,
        left: container.scrollLeft
      };
      var overlayStyle = _extends({}, onELementPosition, customProps.style);
      /* eslint-disable jsx-a11y/no-static-element-interactions */
      return _react2.default.createElement(
        _Portal2.default,
        { isOpen: isOpen, target: container },
        _react2.default.createElement(
          'div',
          {
            ref: function ref(overlay) {
              _this2.overlay = overlay;
            },
            className: overlayClass,
            onClick: this.handleClickOutside,
            style: overlayStyle
          },
          _react2.default.createElement(
            'div',
            {
              tabIndex: 0,
              className: 'terra-Overlay-content'
            },
            children
          )
        )
      );
    }
  }]);

  return Overlay;
}(_react2.default.Component);

Overlay.propTypes = propTypes;
Overlay.defaultProps = defaultProps;

exports.default = Overlay;