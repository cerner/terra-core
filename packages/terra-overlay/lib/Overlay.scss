.terra-Overlay {
  background-color: $terra-dark;
  height: 100%;
  left: 0;
  opacity: 0.8;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: $terra-z-index-4;
}

.terra-Overlay-content {
  max-height: 100%;
  max-width: 100%;
  outline: 0;
  overflow: auto;
  z-index: $terra-z-index-6;
}

.terra-Overlay--center {
  align-items: center;
  display: flex;
  justify-content: center;
}

.terra-Overlay--open {
  overflow: hidden !important;
}
