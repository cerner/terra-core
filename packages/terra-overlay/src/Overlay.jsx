import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import FocusTrap from 'focus-trap-react';
import { Portal } from 'react-portal';
import * as KeyCode from 'keycode-js';
import 'mutationobserver-shim';
import './_contains-polyfill';
import './_matches-polyfill';
import styles from './Overlay.module.scss';
import Container from './OverlayContainer';

const cx = classNames.bind(styles);

const BackgroundStyles = {
  LIGHT: 'light',
  DARK: 'dark',
  CLEAR: 'clear',
};

const zIndexes = ['100', '6000', '7000', '8000', '9000'];

const propTypes = {
  /**
  * The content to be displayed within the overlay.
  */
  children: PropTypes.node,
  /**
  * Indicates if the overlay is open.
  */
  isOpen: PropTypes.bool,
  /**
  * The visual theme to be applied to the overlay background. Accepts 'light', 'dark', and 'clear'.
  */
  backgroundStyle: PropTypes.oneOf(['light', 'dark', 'clear']),
  /**
  * Indicates if the overlay content is scrollable.
  */
  isScrollable: PropTypes.bool,
  /**
  * Callback triggered on overlay click or ESC key. Setting this enables close behavior.
  */
  onRequestClose: PropTypes.func,
  /**
  * Indicates if the overlay is relative to the triggering container.
  */
  isRelativeToContainer: PropTypes.bool,
  /**
   * Used to select the root mount DOM node. This is used to help prevent focus from shifting outside of the overlay when it is opened in a portal.
   */
  rootSelector: PropTypes.string,
  /**
  * Z-Index layer to apply to the ModalContent and ModalOverlay. Valid values are '100', '6000', '7000', '8000', or '9000'.
  */
  zIndex: PropTypes.oneOf(['100', '6000', '7000', '8000', '9000']),
};

const defaultProps = {
  children: null,
  isOpen: false,
  backgroundStyle: BackgroundStyles.LIGHT,
  isScrollable: false,
  isRelativeToContainer: false,
  onRequestClose: undefined,
  rootSelector: '#root',
  zIndex: '100',
};

class Overlay extends React.Component {
  constructor(props) {
    super(props);
    this.setContainer = this.setContainer.bind(this);
    this.disableContainerChildrenFocus = this.disableContainerChildrenFocus.bind(this);
    this.enableContainerChildrenFocus = this.enableContainerChildrenFocus.bind(this);
    this.shouldHandleESCKeydown = this.shouldHandleESCKeydown.bind(this);
    this.shouldHandleClick = this.shouldHandleClick.bind(this);
  }

  componentDidMount() {
    // eslint-disable-next-line no-prototype-builtins
    if (!Element.prototype.hasOwnProperty('inert')) {
      // IE10 throws an error if wicg-inert is imported too early, as wicg-inert tries to set an observer on document.body which may not exist on import
      // eslint-disable-next-line global-require
      require('wicg-inert/dist/inert');
    }
    document.addEventListener('keydown', this.shouldHandleESCKeydown);

    if (this.props.isOpen) {
      this.disableContainerChildrenFocus();
    }
  }

  componentDidUpdate(prevProps) {
    if (this.props.isOpen && !prevProps.isOpen) {
      this.disableContainerChildrenFocus();
    } else if (!this.props.isOpen && prevProps.isOpen) {
      this.enableContainerChildrenFocus();
    }
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.shouldHandleESCKeydown);
    this.enableContainerChildrenFocus();
  }

  setContainer(node) {
    if (!node) { return; } // Ref callbacks happen on mount and unmount, element is null on unmount
    this.overflow = document.documentElement.style.overflow;
    if (this.props.isRelativeToContainer) {
      this.container = node.parentNode;
    } else {
      this.container = null;
    }
  }

  disableContainerChildrenFocus() {
    if (this.props.isRelativeToContainer) {
      if (this.container && this.container.querySelector('[data-terra-overlay-container-content]')) {
        this.container.querySelector('[data-terra-overlay-container-content]').setAttribute('inert', '');
      }
    } else {
      const selector = this.props.rootSelector;

      if (document.querySelector(selector) && !document.querySelector(selector).hasAttribute('data-overlay-count')) {
        document.querySelector(selector).setAttribute('data-overlay-count', '1');
        document.querySelector(selector).setAttribute('inert', '');
      } else if (document.querySelector(selector) && document.querySelector(selector).hasAttribute('data-overlay-count')) {
        const inert = +document.querySelector(selector).getAttribute('data-overlay-count');

        document.querySelector(selector).setAttribute('data-overlay-count', `${inert + 1}`);
        document.querySelector(selector).setAttribute('inert', '');
      }
      document.documentElement.style.overflow = 'hidden';
    }
  }

  enableContainerChildrenFocus() {
    if (this.props.isRelativeToContainer) {
      if (this.container && this.container.querySelector('[data-terra-overlay-container-content]')) {
        this.container.querySelector('[data-terra-overlay-container-content]').removeAttribute('inert');
        this.container.querySelector('[data-terra-overlay-container-content]').removeAttribute('aria-hidden');
      }
    } else {
      const selector = this.props.rootSelector;
      if (document.querySelector(selector)) { // Guard for Jest testing
        const inert = +document.querySelector(selector).getAttribute('data-overlay-count');

        if (inert === 1) {
          document.querySelector(selector).removeAttribute('data-overlay-count');
          document.querySelector(selector).removeAttribute('inert');
          document.querySelector(selector).removeAttribute('aria-hidden');
        } else if (inert && inert > 1) {
          document.querySelector(selector).setAttribute('data-overlay-count', `${inert - 1}`);
        }
      }
      document.documentElement.style.overflow = this.overflow;
    }
  }

  shouldHandleESCKeydown(event) {
    if (this.props.isOpen && event.keyCode === KeyCode.KEY_ESCAPE) {
      this.handleCloseEvent(event);
      event.preventDefault();
    }
  }

  shouldHandleClick(event) {
    if (this.props.isOpen) {
      this.handleCloseEvent(event);
    }
  }

  handleCloseEvent(event) {
    if (this.props.onRequestClose) {
      this.props.onRequestClose(event);
    }
  }

  render() {
    const {
      children, isOpen, backgroundStyle, isScrollable, isRelativeToContainer, onRequestClose, rootSelector, zIndex, ...customProps
    } = this.props;
    const type = isRelativeToContainer ? 'container' : 'fullscreen';

    if (!isOpen) {
      return null;
    }

    let zIndexLayer = '100';
    if (zIndexes.indexOf(zIndex) >= 0) {
      zIndexLayer = zIndex;
    }

    const OverlayClassNames = cx([
      'overlay',
      type,
      backgroundStyle,
      { scrollable: isScrollable },
      customProps.className,
      `layer-${zIndexLayer}`,
    ]);

    /*
      tabIndex set to 0 allows screen readers like VoiceOver to read overlay content when its displayed.
      Key events are added on mount.
    */
    /* eslint-disable jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-tabindex */
    const overlayComponent = (
      <div {...customProps} ref={this.setContainer} onClick={this.shouldHandleClick} className={OverlayClassNames} tabIndex="0">
        <div className={cx('content')}>
          {children}
        </div>
      </div>
    );
    /* eslint-enable jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events, jsx-a11y/no-noninteractive-tabindex */

    if (isRelativeToContainer) {
      return overlayComponent;
    }

    const backgroundScrollContent = (
      <div className={cx('background-scroll-content')}>
        <div className={cx('inner')} />
      </div>
    );

    return (
      <Portal>
        {backgroundScrollContent}
        <FocusTrap>
          {/* div addresses child focus change introduced in focus-trap-react v5.0.0 */}
          <div>
            {overlayComponent}
          </div>
        </FocusTrap>
      </Portal>
    );
  }
}

const Opts = { BackgroundStyles, zIndexes };

Overlay.propTypes = propTypes;
Overlay.defaultProps = defaultProps;
Overlay.Opts = Opts;
Overlay.Container = Container;

export default Overlay;
