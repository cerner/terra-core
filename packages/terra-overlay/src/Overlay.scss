/* stylelint-disable selector-class-pattern  */
// stylelint-disable declaration-property-value-blacklist
:local {
  .overlay-container {
    outline: none;
    position: relative;
  }

  .overlay {
    align-items: center;
    display: flex;
    justify-content: center;
    opacity: 0.8;
    outline: none;
    overflow: hidden;
    z-index: 100;
  }

  .clear {
    background: rgba(255, 255, 255, 0.001);
  }

  .dark {
    background: rgba(28, 31, 33, 0.85);

    .content {
      color: #fff;
    }
  }

  .light {
    background: rgba(244, 244, 244, 0.7);
  }

  .fullscreen {
    bottom: 0;
    left: 0;
    position: fixed;
    right: 0;
    top: 0;
  }

  .container {
    height: 100%;
    position: absolute;
    width: 100%;
  }

  .scrollable {
    overflow-y: scroll;
  }

  .content {
    color: rgba(28, 31, 33, 1);
    font-size: 1rem;
    max-height: 100%;
    padding: 1rem;
  }

  .icon {
    display: block !important;
    margin: auto;
  }

  .message {
    padding: 5px;
  }
}
