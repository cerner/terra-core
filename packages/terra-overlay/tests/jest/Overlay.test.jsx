import React from 'react';
import Overlay from '../../src/Overlay';

describe('Overlay', () => {
  it('should render a null component when isOpen is not provided', () => {
    const wrapper = shallow(<Overlay />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('when isOpen is provided', () => {
    describe('-default Overlay when isOpen-', () => {
      const defaultRender = <Overlay isOpen />;

      it('should render a default component', () => {
        const wrapper = shallow(defaultRender);
        expect(wrapper).toMatchSnapshot();
      });

      it('should have the class overlay', () => {
        const wrapper = shallow(defaultRender);
        expect(wrapper.find('div').at(3).props().className).toContain('overlay');
      });

      it('should default to a fullscreen overlay', () => {
        const wrapper = shallow(defaultRender);
        expect(wrapper.find('div').at(3).props().className).toContain('fullscreen');
      });

      it('should default to 100 zIndex layer', () => {
        const wrapper = shallow(defaultRender);
        expect(wrapper.find('div').at(3).props().className).toContain('layer-100');
      });
    });

    it('should render with content', () => {
      const overlayRender = (
        <Overlay isOpen>
          Overlay message
        </Overlay>
      );
      const wrapper = shallow(overlayRender);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render with isScrollable', () => {
      const overlayRender = <Overlay isOpen isScrollable />;
      const wrapper = shallow(overlayRender);
      expect(wrapper.find('div').at(3).props().className).toContain('scrollable');
      expect(wrapper).toMatchSnapshot();
    });

    it('should render with onRequestClose', () => {
      const overlayRender = <Overlay isOpen onRequestClose={() => {}} />;
      const wrapper = shallow(overlayRender);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render with isRelativeToContainer', () => {
      const overlayRender = <Overlay isOpen isRelativeToContainer />;
      const wrapper = shallow(overlayRender);
      expect(wrapper.prop('className')).toContain('container');
      expect(wrapper).toMatchSnapshot();
    });

    describe('-backgroundStyle-', () => {
      it('should use the default backgroundStyle when no value is given', () => {
        const overlayRender = <Overlay isOpen />;
        const wrapper = shallow(overlayRender);
        expect(wrapper.find('div').at(3).props().className).toContain('light');
      });

      it('should render with the light backgroundStyle', () => {
        const overlayRender = <Overlay isOpen backgroundStyle="light" />;
        const wrapper = shallow(overlayRender);
        expect(wrapper.find('div').at(3).props().className).toContain('light');
        expect(wrapper).toMatchSnapshot();
      });

      it('should render with the dark backgroundStyle', () => {
        const overlayRender = <Overlay isOpen backgroundStyle="dark" />;
        const wrapper = shallow(overlayRender);
        expect(wrapper.find('div').at(3).props().className).toContain('dark');
        expect(wrapper).toMatchSnapshot();
      });

      it('should render with the clear backgroundStyle', () => {
        const overlayRender = <Overlay isOpen backgroundStyle="clear" />;
        const wrapper = shallow(overlayRender);
        expect(wrapper.find('div').at(3).props().className).toContain('clear');
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('-zIndex-', () => {
      it('should render with zIndex 6000', () => {
        const overlayRender = <Overlay isOpen zIndex="6000" />;
        const wrapper = shallow(overlayRender);
        expect(wrapper.find('div').at(3).props().className).toContain('layer-6000');
      });

      it('should render with zIndex 7000', () => {
        const overlayRender = <Overlay isOpen zIndex="7000" />;
        const wrapper = shallow(overlayRender);
        expect(wrapper.find('div').at(3).props().className).toContain('layer-7000');
      });

      it('should render with zIndex 8000', () => {
        const overlayRender = <Overlay isOpen zIndex="8000" />;
        const wrapper = shallow(overlayRender);
        expect(wrapper.find('div').at(3).props().className).toContain('layer-8000');
      });

      it('should render with zIndex 9000', () => {
        const overlayRender = <Overlay isOpen zIndex="9000" />;
        const wrapper = shallow(overlayRender);
        expect(wrapper.find('div').at(3).props().className).toContain('layer-9000');
      });
    });
  });
});
