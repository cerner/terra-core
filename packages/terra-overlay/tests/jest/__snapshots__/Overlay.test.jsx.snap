// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay should render a null component when isOpen is not provided 1`] = `""`;

exports[`Overlay when isOpen is provided -backgroundStyle- should render with the clear backgroundStyle 1`] = `
<Portal>
  <div
    className="background-scroll-content"
  >
    <div
      className="inner"
    />
  </div>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={Object {}}
    paused={false}
  >
    <div>
      <div
        className="overlay fullscreen clear layer-100"
        onClick={[Function]}
        tabIndex="0"
      >
        <div
          className="content"
        />
      </div>
    </div>
  </FocusTrap>
</Portal>
`;

exports[`Overlay when isOpen is provided -backgroundStyle- should render with the dark backgroundStyle 1`] = `
<Portal>
  <div
    className="background-scroll-content"
  >
    <div
      className="inner"
    />
  </div>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={Object {}}
    paused={false}
  >
    <div>
      <div
        className="overlay fullscreen dark layer-100"
        onClick={[Function]}
        tabIndex="0"
      >
        <div
          className="content"
        />
      </div>
    </div>
  </FocusTrap>
</Portal>
`;

exports[`Overlay when isOpen is provided -backgroundStyle- should render with the light backgroundStyle 1`] = `
<Portal>
  <div
    className="background-scroll-content"
  >
    <div
      className="inner"
    />
  </div>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={Object {}}
    paused={false}
  >
    <div>
      <div
        className="overlay fullscreen light layer-100"
        onClick={[Function]}
        tabIndex="0"
      >
        <div
          className="content"
        />
      </div>
    </div>
  </FocusTrap>
</Portal>
`;

exports[`Overlay when isOpen is provided -default Overlay when isOpen- should render a default component 1`] = `
<Portal>
  <div
    className="background-scroll-content"
  >
    <div
      className="inner"
    />
  </div>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={Object {}}
    paused={false}
  >
    <div>
      <div
        className="overlay fullscreen light layer-100"
        onClick={[Function]}
        tabIndex="0"
      >
        <div
          className="content"
        />
      </div>
    </div>
  </FocusTrap>
</Portal>
`;

exports[`Overlay when isOpen is provided should render with content 1`] = `
<Portal>
  <div
    className="background-scroll-content"
  >
    <div
      className="inner"
    />
  </div>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={Object {}}
    paused={false}
  >
    <div>
      <div
        className="overlay fullscreen light layer-100"
        onClick={[Function]}
        tabIndex="0"
      >
        <div
          className="content"
        >
          Overlay message
        </div>
      </div>
    </div>
  </FocusTrap>
</Portal>
`;

exports[`Overlay when isOpen is provided should render with isRelativeToContainer 1`] = `
<div
  className="overlay container light layer-100"
  onClick={[Function]}
  tabIndex="0"
>
  <div
    className="content"
  />
</div>
`;

exports[`Overlay when isOpen is provided should render with isScrollable 1`] = `
<Portal>
  <div
    className="background-scroll-content"
  >
    <div
      className="inner"
    />
  </div>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={Object {}}
    paused={false}
  >
    <div>
      <div
        className="overlay fullscreen light scrollable layer-100"
        onClick={[Function]}
        tabIndex="0"
      >
        <div
          className="content"
        />
      </div>
    </div>
  </FocusTrap>
</Portal>
`;

exports[`Overlay when isOpen is provided should render with onRequestClose 1`] = `
<Portal>
  <div
    className="background-scroll-content"
  >
    <div
      className="inner"
    />
  </div>
  <FocusTrap
    _createFocusTrap={[Function]}
    active={true}
    focusTrapOptions={Object {}}
    paused={false}
  >
    <div>
      <div
        className="overlay fullscreen light layer-100"
        onClick={[Function]}
        tabIndex="0"
      >
        <div
          className="content"
        />
      </div>
    </div>
  </FocusTrap>
</Portal>
`;
